src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts

```diff
@@ -39,6 +47,24 @@ export function useEstimatedGasFee({
     const amount = gasSettings.isEIP1559 ? add(gasSettings.maxBaseFee, gasSettings.maxPriorityFee) : gasSettings.gasPrice;
 
     const totalWei = multiply(gasLimit, amount);
+
+    internalSelectedInputAsset.modify(asset => {
+      'worklet';
+
+      if (asset?.isNativeAsset) {
+        const gasFeeNativeCurrency = divWorklet(totalWei, powWorklet(10, asset.decimals));
+        const gasFeeWithBuffer = toFixedWorklet(mulWorklet(gasFeeNativeCurrency, 1.3), asset.decimals);
```

Comment:
i would define this buffer in the constants.ts file or something. 1.3 hardcoded here is a recipe for misunderstanding later on